src/main/java/bio/overture/rollcall/service/IndexService.java

```diff
@@ -0,0 +1,159 @@
+ *
+ */
+
+package bio.overture.rollcall.service;
+
+import bio.overture.rollcall.index.IndexParser;
+import bio.overture.rollcall.index.ResolvedIndex;
+import bio.overture.rollcall.model.CreateResolvableIndexRequest;
+import bio.overture.rollcall.repository.IndexRepository;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+public class IndexService {
+
+    private final IndexRepository repository;
+    private final AliasService aliasService;
+
+    @Autowired
+    public IndexService(@NonNull IndexRepository repository, @NonNull AliasService aliasService) {
+        this.repository = repository;
+        this.aliasService = aliasService;
+    }
+
+    public List<ResolvedIndex> getResolved() {  return aliasService.getResolved();  }
+
+    public ResolvedIndex createResolvableIndex(@NonNull CreateResolvableIndexRequest createResolvableIndexRequest) {
+        val cloneFromReleasedIndex = createResolvableIndexRequest.getCloneFromReleasedIndex();
+        val newIndexSettings = createResolvableIndexRequest.getIndexSetting();
+
+        // parse request into a dummyResolvedIndex to make sure the parameters are valid, release version is irrelevant
+        val dummyResolvedIndex = generateResolvedIndex(createResolvableIndexRequest, 1);
+
+        // get existing resolved indices with the same alias and shard values
+        val existingResolvedIndices = getRelevantResolvedIndices(dummyResolvedIndex);
+
+        // find resolved index with latest release value in existing resolved indices
+        val latestResolvedIndex = existingResolvedIndices.stream().max(ResolvedIndexByReleaseComparator);
+
+        // create new resolved index, with new release value
+        val newReleaseValue = latestResolvedIndex.isEmpty() ? 1 : calculateNewReleaseValue(latestResolvedIndex.get());
+        val newResolvedIndex = generateResolvedIndex(createResolvableIndexRequest, newReleaseValue);
+        val newIndexName = newResolvedIndex.getIndexName();
+
+        // indexToClone is the published/released index, not the latestResolvedIndex
+        val releasedIndexToClone = findReleasedIndexLikeResolvedIndex(newResolvedIndex);
+
+        if (releasedIndexToClone.isPresent() && cloneFromReleasedIndex) {
+            val indexNameToClone = releasedIndexToClone.get().getIndexName();
+            log.info("Index to clone: " + indexNameToClone);
+            repository.cloneIndex(indexNameToClone, newIndexName, newIndexSettings);
+        } else {
+            repository.createIndex(newIndexName, newIndexSettings);
+        }
+
+        log.info("New cloned/created index name: " + newIndexName);
+        return newResolvedIndex;
+    }
+
+    private Optional<ResolvedIndex> findReleasedIndexLikeResolvedIndex(ResolvedIndex resolvedIndexToCompareWith) {
```

Comment:
Name of this guy is confusing, needs to be clear that this is checking for the currently released (in alias) index with a matching name (entity, type, shardname).

Reply:
Comments describing this would also be fine (since the method is private).